Solutions/Auth0/Data Connectors/Auth0Connector/main.py

```diff
@@ -139,12 +184,42 @@ def _get_token(self):
                 'audience': self.audience
             }
         header = {'content-type': "application/x-www-form-urlencoded"}
-        resp = requests.post(self.domain + '/oauth/token', headers=header, data=params)
-        try:
-            token = resp.json()['access_token']
-        except KeyError:
-            raise Exception('Token not provided.')
+        count = 0
+        error=True
+        while error:
+            try:
+                error=False
+                resp = requests.post(self.domain + '/oauth/token', headers=header, data=params)
+                try:
+                    token = resp.json()['access_token']
+                except KeyError:
+                    raise Exception('Token not provided.')
+            except Exception as err:
+                error = True
+                count+=1
+                if count > self.retry:
+                    break
         return token
 
     def _get_header(self):
         return {'Authorization': 'Bearer ' + self.token}
+    
+    def check_if_script_runs_too_long(self, script_start_time: int) -> bool:
+        now = int(time.time())
+        duration = now - script_start_time
+        max_duration = int(MAX_SCRIPT_EXEC_TIME_MINUTES * 60 * 0.80)
```

Comment:
Can't we move max interval mins to function app paramters

Reply:
Added backlog item to incorporate these changes.